### PR TITLE
[ROCm] Update whisper benchmark script

### PR DESCRIPTION
**Pull request by PeixuanZuo**
- update whisper benchmark for ROCm EP.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/benchmark_helper.py [hunk 1]
```diff
@@ -439,76 +442,132 @@ def get_gpu_info() -> Optional[List[Dict[str, Any]]]:
...
+        except ImportError:
+            self.rocm_smi = None
+
+    def get_used_memory(self, dev):
+        if self.rocm_smi is None:
+            return -1
+        return self.rocm_smi.getMemInfo(dev, "VRAM")[0] / 1024 / 1024
+
+    def measure_gpu_usage(self):
+        device_count = len(self.rocm_smi.listDevices()) if self.rocm_smi is not None else 0
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/benchmark_helper.py:536]: Add a check like this at the beginning of this function.
```
if self.rocm_smi is None:
    return None
```

File: onnxruntime/python/tools/transformers/models/whisper/benchmark.py [hunk 2]
```diff
@@ -352,6 +360,13 @@ def without_io_binding(inputs):
         outputs = model.run(None, inputs)
         return outputs
 
+    def handle_output(output):
+        if 50257 in output:
+            first_end = np.where(output == 50257)[0][0]
+            return output[: first_end + 1]
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/benchmark.py:366]: The end-of-sequence token id is different for the English-only models (e.g. `openai/whisper-tiny.en`). We can add `setattr(args, "eos_token_id", config.eos_token_id)  # noqa: B010` after these lines.

https://github.com/microsoft/onnxruntime/blob/770531d6dc6170fefb29f4e116e88cf4834b9ef0/onnxruntime/python/tools/transformers/models/whisper/benchmark.py#L553-L561

Then we can replace `50257` in `handle_output` with `args.eos_token_id`.

File: onnxruntime/python/tools/transformers/models/whisper/benchmark.py [hunk 3]
```diff
@@ -483,20 +506,29 @@ def parse_args():
     parser.add_argument("--pt-num-rows", type=int, default=1000, help="Number of rows for PyTorch profiler to display")
     parser.add_argument("--verbose", default=False, action="store_true")
     parser.add_argument("--log-folder", type=str, default=os.path.join("."), help="Folder to cache log files")
+    parser.add_argument("--tune", default=False, action="store_true")
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/benchmark.py:509]: Can you add a `help` argument that mentions that `--tune` is for ROCm?

**Comment by kunal-vaishnavi** [comment]
The file permissions on `benchmark_helper.py` appear to have changed. When seeing the file changes, the top of the file says "100644 → 100755".

File: onnxruntime/python/tools/transformers/benchmark_helper.py (hunk 1 above)
Comment (PeixuanZuo) [reply onnxruntime/python/tools/transformers/benchmark_helper.py:539]: Done.

File: onnxruntime/python/tools/transformers/models/whisper/benchmark.py (hunk 3 above)
Comment (PeixuanZuo) [reply onnxruntime/python/tools/transformers/models/whisper/benchmark.py]: Done.

**Review by kunal-vaishnavi — APPROVED** [review]

**Pull request merged**